cmd/kubelet/app/server.go

```diff
@@ -560,8 +560,30 @@ func SimpleKubelet(client *client.Client,
 // Eventually, #2 will be replaced with instances of #3
 func RunKubelet(kcfg *KubeletConfig, builder KubeletBuilder) error {
 	kcfg.Hostname = nodeutil.GetHostname(kcfg.HostnameOverride)
```

Comment:
in `contrib/mesos/pkg/executor/service/service.go` set `kcfg.NodeName = kcfg.Hostname`. also, some unit tests may need updating in `contrib/mesos/pkg/executor/executor_test.go`